server/service/orbit_client.go

```diff
@@ -103,18 +115,31 @@ func NewOrbitClient(
 }
 
 // GetConfig returns the Orbit config fetched from Fleet server for this instance of OrbitClient.
+// Since this method is called in multiple places, we use a cache with configCacheTTL time-to-live to reduce traffic to the Fleet server.
 func (oc *OrbitClient) GetConfig() (*fleet.OrbitConfig, error) {
-	verb, path := "POST", "/api/fleet/orbit/config"
-	var resp orbitGetConfigResponse
-	if err := oc.authenticatedRequest(verb, path, &orbitGetConfigRequest{}, &resp); err != nil {
-		return nil, err
+	oc.configCache.mu.Lock()
+	defer oc.configCache.mu.Unlock()
+	// If time-to-live passed, we update the config cache
+	now := time.Now()
+	if now.After(oc.configCache.lastUpdated.Add(configCacheTTL)) {
+		verb, path := "POST", "/api/fleet/orbit/config"
+		var resp orbitGetConfigResponse
+		if err := oc.authenticatedRequest(verb, path, &orbitGetConfigRequest{}, &resp); err != nil {
+			// We cache the error so that we don't keep hitting the Fleet server if it is having issues
+			oc.configCache.config = nil
+			oc.configCache.err = err
+		} else {
+			oc.configCache.config = &fleet.OrbitConfig{
+				Flags:         resp.Flags,
+				Extensions:    resp.Extensions,
+				Notifications: resp.Notifications,
+				NudgeConfig:   resp.NudgeConfig,
+			}
```

Comment:
how can we prevent people from forgetting to add new fields here? spitballing, can we any variation of this?

```go
oc.configCache.config = resp
```

Reply:
Trying to simply use `OrbitConfig` as resp.